app/src/main/java/com/example/poste/activities/LoginActivity.java

```diff
@@ -1,99 +1,127 @@
-        else {
-            try {
-                User user = API.login(email, password);
-                if (user != null) {
-                    Toast.makeText(LoginActivity.this, getString(R.string.login_successful), Toast.LENGTH_LONG).show();
-                    PosteApplication.setCurrentUser(loginUser);
-                    navigateToDashboard();
-                } else {
-                    Toast.makeText(LoginActivity.this, getString(R.string.login_invalid_credentials), Toast.LENGTH_LONG).show();
-                }
-            } catch (APIException e) {
-                e.printStackTrace();
-                Toast.makeText(LoginActivity.this, getString(R.string.internal_error), Toast.LENGTH_LONG).show();
+        /**
+         * Function that runs when login button is pressed
+         * Makes a request to the API and gets a response in return
+         */
+        buttonLoginSubmit.setOnClickListener(view -> {
+            String email = usernameField.getText().toString();
+            String password = passwordField.getText().toString();
+            if (!validateEmail(LoginActivity.this, email)){
+                return;
             }
+            MyApiService apiService = RetrofitClient.getRetrofitInstance().create(MyApiService.class);
+            Call<ResponseBody> call = apiService.loginUser(new LoginRequest(email, password));
+            call.enqueue(new Callback<ResponseBody>() {
+                /**
+                 * Function that runs when response is returned by the API
+                 * @param call
+                 * @param response
+                 */
+                @Override
+                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                    if (response.isSuccessful()){
+                        Toast.makeText(LoginActivity.this,"Login successful!", Toast.LENGTH_SHORT).show();
+                        try {
+                            String jsonResponse = response.body().string();
+                            JSONObject jsonObject = new JSONObject(jsonResponse);
+                            String token = jsonObject.getJSONObject("result").getString("token");
+                            Log.d("debug","Token: " + token);
+                            User.getUser().setToken(token);
+                            User.getUser().setEmail(email);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        } catch (JSONException err){
+                            Log.e("Json Error", err.toString());
+                        }
+                        // open dashboard activity
+                        Intent dashboardIntent = new Intent(LoginActivity.this, IntroActivity.class);
+                        startActivity(dashboardIntent);
+                    }else {
+                        Toast.makeText(LoginActivity.this, "Login failed.", Toast.LENGTH_SHORT).show();
```

Comment:
May be helpful to parse error responses and display to user; this 'else' blocks indicates a non-success response back from the API, which should contain additional data:

From PosteBackend `LoginView`:
```
        if serializer.is_valid():
            email = serializer.validated_data.get("email")
            password = serializer.validated_data.get("password")
            user = authenticate(request, email=email, password=password)
            if user:
                token, created = Token.objects.get_or_create(user=user)
                return Response(
                    {"result": {"success": True, "token": token.key}},
                    status=status.HTTP_200_OK,
                )

            else:
                return Response(
                    {"result": {"success": False}}, status=status.HTTP_401_UNAUTHORIZED
                )
        else:
            return Response(
                {"result": {"success": False, "errors": serializer.errors}},
                status=status.HTTP_400_BAD_REQUEST,
            )
```

Reply:
Has since been implemented